discovery/kubernetes/node.go

```diff
@@ -205,9 +203,6 @@ func nodeAddress(node *apiv1.Node) (string, map[apiv1.NodeAddressType][]string,
 	if addresses, ok := m[apiv1.NodeExternalIP]; ok {
 		return addresses[0], m, nil
 	}
-	if addresses, ok := m[apiv1.NodeAddressType(api.NodeLegacyHostIP)]; ok {
```

Comment:
IIUC `NodeLegacyHostIP` has been [removed](https://github.com/kubernetes/kubernetes/pull/44830) in Kubernetes 1.7. We need either to update the [documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#node) and warn that it's not supported anymore, or keep it here as a local constant. I'd go with option 2 personally but others will chime in.

Reply:
As we only use APIs that have existed since Kubernetes 1.0 (except for Ingress) I'd prefer keeping compatibility for older versions for the time being until we find an actual benefit of breaking compatibility.

Reply:
Thanks @brancz !